README.md

```diff
@@ -6,106 +6,235 @@
+Creates Zip archives from Node.js or Go source files. Those archives are ready to be uploaded to AWS Lambda.
+
+This library is used under the hood by several Netlify features, including
+[production CI builds](https://github.com/netlify/build), [Netlify CLI](https://github.com/netlify/cli) and the
+[JavaScript client](https://github.com/netlify/js-client).
+
+Check Netlify documentation for:
+
+- [Netlify Functions](https://docs.netlify.com/functions/overview/)
+- [Bundling Functions in the CLI](https://www.netlify.com/docs/cli/#unbundled-javascript-function-deploys)
 
 ## Installation
 
 ```bash
 npm install @netlify/zip-it-and-ship-it
 ```
 
-## Usage
+## API
+
+### zipFunctions(srcFolder, destFolder, options?)
+
+`srcFolder`: `string`\
+`destFolder`: `string`\
+`options`: `object?`\
+_Return value_: `Promise<object[]>`
 
 ```js
 const { zipFunctions } = require('@netlify/zip-it-and-ship-it')
 
-zipFunctions('functions', 'functions-dist')
+const archives = await zipFunctions('functions', 'functions-dist')
 ```
 
-This will take all functions in the `functions` folder and create a matching `.zip` file in the `functions-dist` folder.
+Creates Zip `archives` from Node.js or Go source files. Those `archives` are ready to be uploaded to AWS Lambda.
 
-Each function can either be a single `.js` file that exports a `handler` or a folder with a `.js` with the same name as
-the folder exporting a handler.
+`srcFolder` is the source files directory. It must exist. In Netlify, this is the
+["Functions folder"](https://docs.netlify.com/functions/configure-and-deploy/#configure-the-functions-folder).
 
-The packaging tool will look for the `package.json` closest to the handler and use that for dependency resolution. Make
-sure you've run `npm install` or `yarn` for each `package.json` before using `zip-it-and-ship-it`.
+`srcFolder` can contain:
 
-Ie, the following combinations would all work:
+- Sub-directories with a main file called either `index.js` or `{dir}.js` where `{dir}` is the sub-directory name.
+- `.js` files (Node.js)
+- `.zip` archives with Node.js already ready to upload to AWS Lambda.
+- Go files, already built. If the `options.zipGo` is `true`, those are zipped. Otherwise, those are copied as is.
 
-```console
-/functions/foo.js
-/package.json
-/node_modules/
-```
+When using Node.js files, only the dependencies required by the main file are bundled, in order to keep the archive as
+small as possible, which improves the Function runtime performance:
 
-```console
-/functions/foo.js
-/functions/bar/bar.js
-/functions/package.json
-/functions/node_modules/
-```
+- All files/directories within the same directory (except `node_modules`) are included
+- All the `require()`'d files are included
+- All the `require()`'d `node_modules` are included, recursively
+- The following modules are never included:
+  - `@types/*` TypeScript definitions
+  - `aws-sdk`
```

Comment:
Not relevant for the readme, but I think it is best practice to bundle `aws-sdk`:
https://theburningmonk.com/2019/09/should-you-pack-the-aws-sdk-in-your-deployment-artefact/

Reply:
Hum, interesting.

That part of the code was there before I join in, so I am not completely sure why it was there. It looks like it was part of the repo's first commit, created by @biilmann. @biilmann, do you have some additional insights on this?

Reply:
I think the purpose is to minimize the bundle size as NodeJS lambda runtime comes with its own version of `aws-sdk`